cmd/clusterctl/client/repository/repository_github_test.go

```diff
@@ -30,11 +30,204 @@ import (
 	"sigs.k8s.io/cluster-api/cmd/clusterctl/internal/test"
 )
 
-//TODO: test newGitHubRepository
+func Test_githubRepository_newGitHubRepository(t *testing.T) {
+	g := NewWithT(t)
+
+	type field struct {
+		providerConfig config.Provider
+		variableClient config.VariablesClient
+	}
+	tests := []struct {
+		name    string
+		field   field
+		want    *gitHubRepository
+		wantErr bool
+	}{
+		{
+			name: "can create a new GitHub repo",
+			field: field{
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+				variableClient: test.NewFakeVariableClient(),
+			},
+			want: &gitHubRepository{
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+
+				configVariablesClient:    test.NewFakeVariableClient(),
+				authenticatingHTTPClient: nil,
+				owner:                    "o",
+				repository:               "r1",
+				defaultVersion:           "v0.4.1",
+				rootPath:                 ".",
+				componentsPath:           "path",
+				injectClient:             nil,
+			},
+			wantErr: false,
+		},
+		{
+			name: "missing variableClient",
+			field: field{
+				providerConfig: config.NewProvider("test", "https://github.com/o/r1/releases/v0.4.1/path", clusterctlv1.CoreProviderType),
+				variableClient: nil,
+			},
+			want:    &gitHubRepository{},
+			wantErr: true,
+		},
+		{
+			name: "provider url is not valid",
+			field: field{
+				providerConfig: config.NewProvider("test", "%gh&%ij", clusterctlv1.CoreProviderType),
+				variableClient: test.NewFakeVariableClient(),
+			},
+			want:    &gitHubRepository{},
+			wantErr: true,
+		},
+		{
+			name: "provider url should be in https",
+			field: field{
+				providerConfig: config.NewProvider("test", "http://github.com/blabla", clusterctlv1.CoreProviderType),
+				variableClient: test.NewFakeVariableClient(),
+			},
+			want:    &gitHubRepository{},
+			wantErr: true,
+		},
+		{
+			name: "provider url should be in github",
+			field: field{
+				providerConfig: config.NewProvider("test", "http://gitlab.com/blabla", clusterctlv1.CoreProviderType),
+				variableClient: test.NewFakeVariableClient(),
+			},
+			want:    &gitHubRepository{},
+			wantErr: true,
+		},
+		{
+			name: "provider url should be in https://github.com/{owner}/{Repository}/%s/{latest|version-tag}/{componentsClient.yaml} format",
+			field: field{
+				providerConfig: config.NewProvider("test", "https://github.com/dd/", clusterctlv1.CoreProviderType),
+				variableClient: test.NewFakeVariableClient(),
+			},
+			want:    &gitHubRepository{},
+			wantErr: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gitHub, err := newGitHubRepository(tt.field.providerConfig, tt.field.variableClient)
```

Comment:
```suggestion
		t.Run(tt.name, func(t *testing.T) {
		        g := NewWithT(t)
			gitHub, err := newGitHubRepository(tt.field.providerConfig, tt.field.variableClient)
```
Move the `g := NewWithT(t)` from line 34 here to pass in the correct context to ginkgo.

Currently, the errors throw
```
    --- FAIL: Test_githubRepository_newGitHubRepository/can_create_a_new_GitHub_repo (0.00s)
        testing.go:864: test executed panic(nil) or runtime.Goexit: subtest may have called FailNow on a parent test
```